### ISSUE TITLE
Training details for BAR + OAT + PaliGemma2

### ISSUE THREAD

**Issue by Evan-XLiu** [Chaoqi-LIU/praxis-vla#2]
Hi, thank you very much for releasing Praxis and OAT！ 
I am currently using praxis-vla almost entirely as my training pipeline, and I have a question about the training recipe for bar_oat_pali.

I am adapting the pipeline to RoboTwin with approximately 2,500 demonstrations. My current training procedure is:

Train an OAT action tokenizer on all ~2,500(50 tasks x 50 demos) RoboTwin demonstrations. (300000steps finished, reconst_loss is around 10-3).
Freeze the pretrained OAT tokenizer.
Train BAR + OAT + PaliGemma2 using google/paligemma2-3b-mix-224.
I use full fine-tuning (use_lora=false) and do not freeze the vision encoder (freeze_vision_encoder=false).
The policy has currently been trained for about 2,600 optimizer steps(reconst_loss is around 10-2).

<img width="675" height="269" alt="Image" src="https://github.com/user-attachments/assets/4c115d24-da5d-4817-8d2e-c3b91b5e12c6" />

The validation reconstruction loss has already converged reasonably well, but the closed-loop RoboTwin performance is still very poor, with almost no successful rollouts.

https://github.com/user-attachments/assets/59cbd3c3-2682-4b21-b69e-2304be77f4e9

I would really appreciate some clarification on the training details used for the VLM-scale BAR experiments in the paper:

**Roughly how many optimizer steps / epochs were used for the PaliGemma2 + BAR + OAT experiments?
Is ~2.6k steps expected to be far too short, even when the offline reconstruction loss has already converged? Table 8 of the paper reports a batch size of 16 for AR policies, while the current generic trainer config appears to use a per-rank batch size of 64. Could you clarify the exact effective/global batch size used for the reported PaliGemma2 experiments?**

In your experience, is val_reconst_loss a useful proxy for selecting BAR checkpoints, or do you primarily rely on closed-loop success rate? I am seeing a large gap between converged reconstruction loss and rollout success.
Are there any particularly important implementation details when adapting Praxis to a new bimanual dataset, e.g. action normalization, proprioceptive state encoding, action ordering, or action execution horizon, that could cause good offline reconstruction but very poor rollout performance?

If possible, it would also be extremely helpful to have one reference Hydra/launch configuration corresponding to a reported PaliGemma2 + BAR + OAT experiment.

Thanks again for the excellent work and for open-sourcing the framework!

**Comment by Chaoqi-LIU** [comment]
reference: [libero+bar+oat+qwen](https://drive.google.com/drive/folders/116u0Pqu4-qeicE7GsoHOlE5mWgs7cj1P?usp=sharing) & its [tokenizer](https://drive.google.com/drive/folders/1HDtIgRBM6wga6D21IttDrtzcKWTEbNs7?usp=sharing). This includes everything, checkpoint, wandb, eval results, config, and more. But this is artifacts during development, and code entry point has changed since then, I don't expect it can be loaded and use immediately without adaption. But I'm sure Astra/Fable would do this.

I use global batch size 16 for BAR results, and target 500k steps, while most of them perform best around 100k-150k, I do eval every 10k/20k, so I can pick the best based on success rate. If not able to eval frequently, try follow reconst MSE, and most of the them, these specialist policy on one benchmark, work best when it starts to overfit, i.e., see increase in val loss.

**Comment by Chaoqi-LIU** [comment]
feel free to reach out via email, I may failed to respond to new github issues promptly.
